src/compiler/utilities.ts

```diff
@@ -6292,8 +6292,18 @@ namespace ts {
 
     /** Returns true for the first identifier of 1) an `implements` clause, and 2) an `extends` clause of an interface. */
     function isFirstIdentifierOfNonEmittingHeritageClause(node: Node): boolean {

```

Comment:
I don't get where "first"-ness plays into the implementation here, or why you'd want it. If you're only looking the first part of an `extends` or `implements`, does this mean that the following

```ts
import type { Baz } from "whatever

interface Bar {}

interface Foo extends Bar, Baz {}
```

still doesn't work?

Reply:
It’s not the first-ness of the heritage clause, it’s the first-ness of the identifier that we’re looking at. Which also doesn’t strictly matter, but this function will never run on something in the middle/end of a property access expression, so originally it was intended to be a performance enhancement to bail early on those. But, that was, again, written with an inverted tree in mind (🤦‍♂ 🤦‍♂ 🤦‍♂), so now this is in fact a misnomer, though the logic is good. I’ll rename it.